src/west/manifest.py

```diff
@@ -857,17 +859,25 @@ def _import_path_from_project(self, project, path, projects):
 
             for i, subp in enumerate(submp.projects):
                 if i == MANIFEST_PROJECT_INDEX:
-                    # If the project has no west commands, inherit them
-                    # from imported manifest data inside the project.
-                    if not has_wc and subp.west_commands:
-                        if isinstance(subp.west_commands, str):
-                            inherited_wc.append(subp.west_commands)
-                        else:
-                            inherited_wc.extend(subp.west_commands)
+                    # If the submanifest has west commands, add them
+                    # to project. Try to keep project.west_commands a
+                    # str, but upgrade it to a list if there's no
+                    # choice.
+                    subcmds = subp.west_commands
```

Comment:
Would it make sense to create a function to avoid the duplication of [this code](https://github.com/zephyrproject-rtos/west/pull/354/commits/758ff0def814ac33b2418d21f4b511a8573f1fa7#diff-aa9e31a9849c50f6ad6cfc1919031fa1R693)?

Reply:
Yeah, I was just being lazy.

Reply:
Now made a function; will merge once CI passes.